wpimath/src/test/java/edu/wpi/first/math/estimator/PoseEstimationTestUtils.java

```diff
@@ -0,0 +1,69 @@
+package edu.wpi.first.math.estimator;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Twist2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.kinematics.Kinematics;
+import edu.wpi.first.math.kinematics.WheelPositions;
+
+public class PoseEstimationTestUtils {
+  /**
+   * This class is intended for use as "drivetrain-agnostic" kinematics. 
+   * Odometry measurements are akin to Poses, while their difference is akin to Twists.
+   */
+  protected static class SE2KinematicPrimitive implements WheelPositions<SE2KinematicPrimitive> {
+    private Pose2d pose;
+
+    public SE2KinematicPrimitive(final Pose2d pose) {
+      this.pose = pose;
+    }
+
+    @Override
+    public SE2KinematicPrimitive copy() {
+      return this;
+    }
+
+    @Override
+    public SE2KinematicPrimitive minus(SE2KinematicPrimitive other) {
+      return new SE2KinematicPrimitive(new Pose2d().exp(other.pose.log(this.pose)));
```

Comment:
Minor: As outlined in the comment on `PoseEstimator.sampleAt`, `a.exp(b.log(c))` is equivalent to `a + (c - b)`, so this can be simplified to `new Pose2d().plus(this.pose.minus(other.pose))`, or `this.pose.relativeTo(other.pose)`. There's one or two other spots that could also be changed, but it doesn't matter much- It just saves a few calculations.

Reply:
exp() and plus() are not the same operations.

Reply:
I think this has become irrelevant? More recent changes removed the minus() function in favor of implementing everything in kinematics classes.